core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeInfo.java

```diff
@@ -44,29 +43,27 @@
    * which the request was made, otherwise the new node will be ignored.
    */
   InetSocketAddress getConnectAddress();
-
   /**
    * The node's broadcast address. That is, the address that other nodes use to communicate with
    * that node.
    *
    * <p>This is only used by the default topology monitor, so if you are writing a custom one and
    * don't need this information, you can leave it empty.
    */
-  Optional<InetAddress> getBroadcastAddress();
+  Optional<InetSocketAddress> getBroadcastAddress();
 
   /**
    * The node's listen address. That is, the address that the Cassandra process binds to.
    *
    * <p>This is currently not used anywhere in the driver. If you write a custom topology monitor
    * and don't need this information, you can leave it empty.
    */
-  Optional<InetAddress> getListenAddress();
-
+  Optional<InetSocketAddress> getListenAddress();
   /**
-   * The data center that this node belongs to, according to the Cassandra snitch.
+   * The node's listen port. That is, the port that the Cassandra process binds to.
    *
-   * <p>This is used by some {@link LoadBalancingPolicy} implementations to compute the {@link
-   * NodeDistance}.
+   * <p>This is currently not used anywhere in the driver. If you write a custom topology monitor
+   * and don't need this information, you can leave it empty.
```

Comment:
Mmm I disagree, `getDatacenter()` is still used by the default LBP. Not exactly for distance, but non-local nodes are ignored based on this information.